### DISCUSSION TITLE
No coupler.res and sfc.nc files when restarting a warm start or restart run

### DISCUSSION THREAD

**Discussion by ericaligo-NOAA** [ufs-community/ufs-weather-model#2402] [Q&A] [answered]
I'm running FV3 off of warm start ICs/LBCs from the RRFS. When I attempted to restart this run, I did not see a coupler.res or sfc.nc file in the RESTART run directory. Is this to be expected?

**Comment by gspetro-NOAA** [comment]
Hi @ericaligo-NOAA ,

It looks like the baseline case (`conus13km_control`) does require restart files. @jkbk2004 has suggested checking your configuration against the [baseline conus13km_control case](https://github.com/ufs-community/ufs-weather-model/blob/develop/tests/tests/conus13km_control#L20-L26) to see what might be awry. If you run off of the production/RRFS.v1 branch, you could check your configuration against the baseline cases in [rt.conf_rrfs](https://github.com/ufs-community/ufs-weather-model/blob/production/RRFS.v1/tests/rt.conf_rrfs). If that does not clarify the issue, we'd suggest reaching out to @MatthewPyle-NOAA for advice. 

Best,
Gillian Petro

**Reply by gspetro-NOAA** [reply]
Hi @ericaligo-NOAA ,

Just wanted to check in and see if you need any further assistance or whether I can close this discussion. 

Best,
Gillian

**Reply by gspetro-NOAA** [reply]
Closing discussion, since no further need for assistance has been indicated. Feel free to open a new discussion or request that this one be reopened if needed.

**Comment by ericaligo-NOAA** [comment]
I had responded via email, and I don't know why my emails don't show up hear. I did need further assistance. Here's the email I just sent:

I would like further assistance. When I do a restart run at f03, how do I enable f03 to be regenerated? Right now, the first file I see in a restart run with hourly output, is f04. In the past, the first file generated in a restart run would have f03.

**Reply by gspetro-NOAA** [reply]
Hi @ericaligo-NOAA ,

It seems like if you are doing a restart run from f03, then you would already have the restart info for f03. Am I misunderstanding? What are you trying to get from f03 that isn't already in the f03 restart file you're using? 

Best,
Gillian

**Comment by ericaligo-NOAA** [comment]
Hi,
Code changes were made such that they would be reflected in the first time step of the restart. In the past, if I asked for a restart at f03, f03 would be regenerated, however, that file is now missing, and the first file I see is f04. I need to compare the restarted f03 file with the original f03 forecast file.

**Comment by jkbk2004** [comment]
@ericaligo-NOAA Regarding your request, only option can be along with https://github.com/NOAA-EMC/fv3atm/blob/a9364591091c836984a40107729720705847c195/module_fcst_grid_comp.F90#L796-L810. But I don't think restart file re-generation is an option. @junwang-noaa @DusanJovic-NOAA can you confirm there is an option of restart file write-out at start time?

**Comment by ericaligo-NOAA** [comment]
Hi, this was an option up until recently. I didn't know if a namelist variable was added and I need to set it in order to turn on this feature.

**Comment by DusanJovic-NOAA** [comment, marked as answer]
I asked Jun, and this was her response: "I think we can only output restart files at the first model step. Please set first time step (could be real) as restart_interval in model_configure."
